<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>GeneBridge</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="index_files/libs/clipboard/clipboard.min.js"></script>
<script src="index_files/libs/quarto-html/quarto.js"></script>
<script src="index_files/libs/quarto-html/popper.min.js"></script>
<script src="index_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="index_files/libs/quarto-html/anchor.min.js"></script>
<link href="index_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="index_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="index_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="index_files/libs/bootstrap/bootstrap-8a79a254b8e706d3c925cde0a310d4f0.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#genebridge" id="toc-genebridge" class="nav-link active" data-scroll-target="#genebridge"><span class="header-section-number">1</span> GeneBridge</a>
  <ul class="collapse">
  <li><a href="#importando-pacotes" id="toc-importando-pacotes" class="nav-link" data-scroll-target="#importando-pacotes"><span class="header-section-number">1.1</span> Importando Pacotes</a></li>
  <li><a href="#definindo-funções-para-usar-posteriormente" id="toc-definindo-funções-para-usar-posteriormente" class="nav-link" data-scroll-target="#definindo-funções-para-usar-posteriormente"><span class="header-section-number">1.2</span> Definindo funções para usar posteriormente</a></li>
  <li><a href="#carregando-lista-de-genes-e-os-dados-de-ortologia" id="toc-carregando-lista-de-genes-e-os-dados-de-ortologia" class="nav-link" data-scroll-target="#carregando-lista-de-genes-e-os-dados-de-ortologia"><span class="header-section-number">1.3</span> Carregando lista de genes e os dados de ortologia</a></li>
  <li><a href="#pré-processamento" id="toc-pré-processamento" class="nav-link" data-scroll-target="#pré-processamento"><span class="header-section-number">1.4</span> 1.Pré-processamento</a>
  <ul class="collapse">
  <li><a href="#mapeamento" id="toc-mapeamento" class="nav-link" data-scroll-target="#mapeamento"><span class="header-section-number">1.4.1</span> 1.1.Mapeamento</a></li>
  <li><a href="#resolverndo-cogs-duplicados" id="toc-resolverndo-cogs-duplicados" class="nav-link" data-scroll-target="#resolverndo-cogs-duplicados"><span class="header-section-number">1.4.2</span> 1.2.Resolverndo COGs duplicados</a></li>
  </ul></li>
  <li><a href="#processamento" id="toc-processamento" class="nav-link" data-scroll-target="#processamento"><span class="header-section-number">1.5</span> 3.Processamento</a>
  <ul class="collapse">
  <li><a href="#inputs-necessários" id="toc-inputs-necessários" class="nav-link" data-scroll-target="#inputs-necessários"><span class="header-section-number">1.5.1</span> 3.1.Inputs necessários</a></li>
  </ul></li>
  <li><a href="#pós-processamento" id="toc-pós-processamento" class="nav-link" data-scroll-target="#pós-processamento"><span class="header-section-number">1.6</span> 4.Pós-Processamento</a>
  <ul class="collapse">
  <li><a href="#rede-de-interação-proteína-proteína" id="toc-rede-de-interação-proteína-proteína" class="nav-link" data-scroll-target="#rede-de-interação-proteína-proteína"><span class="header-section-number">1.6.1</span> 4.1.Rede de Interação Proteína-Proteína</a></li>
  </ul></li>
  <li><a href="#importar-bibliotecas" id="toc-importar-bibliotecas" class="nav-link" data-scroll-target="#importar-bibliotecas"><span class="header-section-number">1.7</span> Importar bibliotecas</a></li>
  <li><a href="#definir-funções" id="toc-definir-funções" class="nav-link" data-scroll-target="#definir-funções"><span class="header-section-number">1.8</span> Definir funções</a></li>
  <li><a href="#carregando-tabelas-necessárias" id="toc-carregando-tabelas-necessárias" class="nav-link" data-scroll-target="#carregando-tabelas-necessárias"><span class="header-section-number">1.9</span> Carregando tabelas necessárias</a></li>
  <li><a href="#visualização-com-upset-plot" id="toc-visualização-com-upset-plot" class="nav-link" data-scroll-target="#visualização-com-upset-plot"><span class="header-section-number">1.10</span> 1.Visualização com UpSet Plot</a></li>
  <li><a href="#visualização-da-rede-de-interação-proteína-proteína" id="toc-visualização-da-rede-de-interação-proteína-proteína" class="nav-link" data-scroll-target="#visualização-da-rede-de-interação-proteína-proteína"><span class="header-section-number">1.11</span> 2.Visualização da Rede de Interação Proteína-Proteína</a>
  <ul class="collapse">
  <li><a href="#visualização-da-ancestralidade-de-cada-nó" id="toc-visualização-da-ancestralidade-de-cada-nó" class="nav-link" data-scroll-target="#visualização-da-ancestralidade-de-cada-nó"><span class="header-section-number">1.11.1</span> 2.1.Visualização da Ancestralidade de Cada Nó</a></li>
  <li><a href="#visualização-da-rede-de-interação-proteína-proteína-em-humano" id="toc-visualização-da-rede-de-interação-proteína-proteína-em-humano" class="nav-link" data-scroll-target="#visualização-da-rede-de-interação-proteína-proteína-em-humano"><span class="header-section-number">1.11.2</span> 2.2.Visualização da Rede de Interação Proteína-Proteína em Humano</a></li>
  <li><a href="#visualização-da-rede-de-interação-proteína-proteína-em-cada-clado" id="toc-visualização-da-rede-de-interação-proteína-proteína-em-cada-clado" class="nav-link" data-scroll-target="#visualização-da-rede-de-interação-proteína-proteína-em-cada-clado"><span class="header-section-number">1.11.3</span> 2.3.Visualização da Rede de Interação Proteína-Proteína em Cada Clado</a></li>
  </ul></li>
  <li><a href="#importando-pacotes-1" id="toc-importando-pacotes-1" class="nav-link" data-scroll-target="#importando-pacotes-1"><span class="header-section-number">1.12</span> Importando Pacotes</a></li>
  <li><a href="#definindo-funções" id="toc-definindo-funções" class="nav-link" data-scroll-target="#definindo-funções"><span class="header-section-number">1.13</span> Definindo funções</a></li>
  <li><a href="#definindo-parâmetros-estéticos-para-o-ggplot" id="toc-definindo-parâmetros-estéticos-para-o-ggplot" class="nav-link" data-scroll-target="#definindo-parâmetros-estéticos-para-o-ggplot"><span class="header-section-number">1.14</span> Definindo parâmetros estéticos para o ggplot</a></li>
  <li><a href="#carregar-tabelas-necessárias" id="toc-carregar-tabelas-necessárias" class="nav-link" data-scroll-target="#carregar-tabelas-necessárias"><span class="header-section-number">1.15</span> Carregar tabelas necessárias</a>
  <ul class="collapse">
  <li><a href="#descrição-da-tabela" id="toc-descrição-da-tabela" class="nav-link" data-scroll-target="#descrição-da-tabela"><span class="header-section-number">1.15.1</span> Descrição da Tabela</a></li>
  </ul></li>
  <li><a href="#visualização-padrão-de-enraizamento" id="toc-visualização-padrão-de-enraizamento" class="nav-link" data-scroll-target="#visualização-padrão-de-enraizamento"><span class="header-section-number">1.16</span> Visualização: Padrão de Enraizamento</a>
  <ul class="collapse">
  <li><a href="#características-dos-gráficos" id="toc-características-dos-gráficos" class="nav-link" data-scroll-target="#características-dos-gráficos"><span class="header-section-number">1.16.1</span> Características dos Gráficos</a></li>
  <li><a href="#interpretação" id="toc-interpretação" class="nav-link" data-scroll-target="#interpretação"><span class="header-section-number">1.16.2</span> Interpretação</a></li>
  </ul></li>
  <li><a href="#cálculo-da-abundância-de-cogs-por-função-nas-espécies" id="toc-cálculo-da-abundância-de-cogs-por-função-nas-espécies" class="nav-link" data-scroll-target="#cálculo-da-abundância-de-cogs-por-função-nas-espécies"><span class="header-section-number">1.17</span> Cálculo da Abundância de COGs por Função nas Espécies</a>
  <ul class="collapse">
  <li><a href="#etapas-do-cálculo" id="toc-etapas-do-cálculo" class="nav-link" data-scroll-target="#etapas-do-cálculo"><span class="header-section-number">1.17.1</span> Etapas do Cálculo</a></li>
  <li><a href="#visualização" id="toc-visualização" class="nav-link" data-scroll-target="#visualização"><span class="header-section-number">1.17.2</span> Visualização</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">GeneBridge</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="genebridge" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> GeneBridge</h1>
<section id="importando-pacotes" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="importando-pacotes"><span class="header-section-number">1.1</span> Importando Pacotes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GeneBridge)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(geneplast.data)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(KEGGREST)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ape)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringi)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sourcetools)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="definindo-funções-para-usar-posteriormente" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="definindo-funções-para-usar-posteriormente"><span class="header-section-number">1.2</span> Definindo funções para usar posteriormente</h2>
<p>A primeira delas, busca o respectivo id de proteína para cada gene da lista de entrada. A segunda, retorna as interações entre essas proteínas e a última filtra as interações pelo score de confiança combinado maior que 0.4.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get IDs from STRING DB</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>get_string_ids <span class="ot">&lt;-</span> <span class="cf">function</span>(genes_hgnc, <span class="at">species_id =</span> <span class="st">"9606"</span>) {</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>    req <span class="ot">&lt;-</span> RCurl<span class="sc">::</span><span class="fu">postForm</span>(</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"https://string-db.org/api/tsv/get_string_ids"</span>,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">identifiers =</span> <span class="fu">paste</span>(genes_hgnc, <span class="at">collapse =</span> <span class="st">"%0D"</span>),  </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">echo_query =</span> <span class="st">"1"</span>,</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> species_id, </span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">.opts =</span> <span class="fu">list</span>(<span class="at">ssl.verifypeer =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  map_ids <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="at">text =</span> req, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">quote =</span> <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>queryIndex) <span class="sc">%&gt;%</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>()</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  map_ids<span class="sc">$</span>stringId <span class="ot">&lt;-</span> <span class="fu">substring</span>(map_ids<span class="sc">$</span>stringId, <span class="dv">6</span>, <span class="dv">1000</span>)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(map_ids)</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a> <span class="co"># Get STRING interactions</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a> get_network_interaction <span class="ot">&lt;-</span> <span class="cf">function</span>(map_ids, protein_id, <span class="at">species_id =</span> <span class="st">"9606"</span>) {</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>   identifiers <span class="ot">&lt;-</span> map_ids <span class="sc">%&gt;%</span> <span class="fu">pull</span>(protein_id) <span class="sc">%&gt;%</span> na.omit <span class="sc">%&gt;%</span> <span class="fu">paste0</span>(<span class="at">collapse=</span><span class="st">"%0d"</span>) </span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>   req2 <span class="ot">&lt;-</span> RCurl<span class="sc">::</span><span class="fu">postForm</span>(</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>     <span class="st">"https://string-db.org/api/tsv/network"</span>,</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>     <span class="at">identifiers =</span> identifiers, </span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>     <span class="at">required_core =</span> <span class="st">"0"</span>, </span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>     <span class="at">species =</span> species_id,</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>     <span class="at">.opts =</span> <span class="fu">list</span>(<span class="at">ssl.verifypeer =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>   )</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>   int_network <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="at">text =</span> req2, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>   int_network <span class="ot">&lt;-</span> <span class="fu">unique</span>(int_network)</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>   <span class="fu">return</span>(int_network)</span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a> <span class="do">## Recomputing scores</span></span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>combine_scores <span class="ot">&lt;-</span> <span class="cf">function</span>(dat, <span class="at">evidences =</span> <span class="st">"all"</span>, <span class="at">confLevel =</span> <span class="fl">0.4</span>) {</span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(evidences[<span class="dv">1</span>] <span class="sc">==</span> <span class="st">"all"</span>){</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>    edat<span class="ot">&lt;-</span>dat[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="fu">ncol</span>(dat))]</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">all</span>(evidences<span class="sc">%in%</span><span class="fu">colnames</span>(dat))){</span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stop</span>(<span class="st">"NOTE: one or more 'evidences' not listed in 'dat' colnames!"</span>)</span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>    edat<span class="ot">&lt;-</span>dat[,evidences]</span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">any</span>(edat <span class="sc">&gt;</span> <span class="dv">1</span>)) {</span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>    edat <span class="ot">&lt;-</span> edat<span class="sc">/</span><span class="dv">1000</span></span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>  edat<span class="ot">&lt;-</span><span class="dv">1</span><span class="sc">-</span>edat</span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>  sc<span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="at">X =</span> edat, <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">FUN =</span> <span class="cf">function</span>(x) <span class="dv">1</span><span class="sc">-</span><span class="fu">prod</span>(x))</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>  dat <span class="ot">&lt;-</span> <span class="fu">cbind</span>(dat[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)],<span class="at">combined_score =</span> sc)</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>  idx <span class="ot">&lt;-</span> dat<span class="sc">$</span>combined_score <span class="sc">&gt;=</span> confLevel</span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>  dat <span class="ot">&lt;-</span>dat[idx,]</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(dat)</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="carregando-lista-de-genes-e-os-dados-de-ortologia" class="level2" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="carregando-lista-de-genes-e-os-dados-de-ortologia"><span class="header-section-number">1.3</span> Carregando lista de genes e os dados de ortologia</h2>
<p>Carregamos os dados de ortologia através do <strong>AnnotationHub</strong>, esse pacote do R fornece um local central onde arquivos genômicos (VCF, bed, wig) e outros recursos de locais padrões (por exemplo, UCSC, Ensembl) podem ser acessados. Dessa forma, temos acesso aos arquivos de entrada para o algoritmo do GeneBridge.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the Gene Set Table</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>sensorial_genes <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/sensorial_genes.csv"</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Query Phylotree and OG data</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>ah <span class="ot">&lt;-</span> <span class="fu">AnnotationHub</span>()</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">query</span>(ah, <span class="st">"geneplast"</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(meta[[<span class="st">"AH83116"</span>]])</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sensorial_genes)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cogdata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pré-processamento" class="level2" data-number="1.4">
<h2 data-number="1.4" class="anchored" data-anchor-id="pré-processamento"><span class="header-section-number">1.4</span> 1.Pré-processamento</h2>
<section id="mapeamento" class="level3" data-number="1.4.1">
<h3 data-number="1.4.1" class="anchored" data-anchor-id="mapeamento"><span class="header-section-number">1.4.1</span> 1.1.Mapeamento</h3>
<p>Para as próximas análises, precisamos cruzar informações entre nossos genes de interesse (Gene IDs da tabela <strong><code>sensorial_genes</code></strong>) e Protein IDs (da tabela <strong><code>cogdata</code></strong>). A API do STRINGdb é usada para mapear os Gene IDs para os Protein IDs, permitindo a filtragem dos genes de interesse na tabela cogdata. O objetivo final é obter um conjunto filtrado de genes sensoriais com seus respectivos pathways e COG IDs.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>map_ids <span class="ot">&lt;-</span> <span class="fu">get_string_ids</span>(sensorial_genes<span class="sc">$</span>gene_symbol)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Subsetting cogs of interest - Sensorial Genes</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>gene_cogs <span class="ot">&lt;-</span> cogdata <span class="sc">%&gt;%</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ssp_id <span class="sc">%in%</span> map_ids<span class="sc">$</span>ncbiTaxonId) <span class="sc">%&gt;%</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(protein_id <span class="sc">%in%</span> map_ids[[<span class="st">"stringId"</span>]]) <span class="sc">%&gt;%</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(protein_id) <span class="sc">%&gt;%</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>(), <span class="at">cog_id =</span> <span class="fu">paste</span>(cog_id, <span class="at">collapse =</span> <span class="st">" / "</span>))</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(map_ids)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co">#map_ids |&gt; </span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">#  vroom::vroom_write(file = here("data/map_ids.csv"), delim = ",")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="resolverndo-cogs-duplicados" class="level3" data-number="1.4.2">
<h3 data-number="1.4.2" class="anchored" data-anchor-id="resolverndo-cogs-duplicados"><span class="header-section-number">1.4.2</span> 1.2.Resolverndo COGs duplicados</h3>
<p>Devido a eventos evolutivos, como duplicação gênica, alguns genes podem ser associados a mais de um Cluster of Orthologous Groups (COG). Para garantir a funcionalidade do algoritmo, é necessário resolver esses casos, priorizando COGs de acordo com os seguintes critérios:</p>
<ol type="1">
<li>Prioridade por tipo de COG:</li>
</ol>
<ul>
<li>KOGs têm maior prioridade.</li>
<li>COGs têm maior prioridade do que NOGs.</li>
</ul>
<ol start="2" type="1">
<li>Casos com COGs iniciando pela mesma letra:</li>
</ol>
<ul>
<li>São resolvidos manualmente, com base na função anotada do COG e na questão científica do estudo.</li>
</ul>
<p>O código abaixo implementa essa resolução e integra as correções à tabela principal.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>gene_cogs <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Resolving main proteins</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>gene_cogs_resolved <span class="ot">&lt;-</span> <span class="fu">tribble</span>(</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span>protein_id, <span class="sc">~</span>cog_id,</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="st">"ENSP00000332500"</span>, <span class="st">"NOG274749"</span>, <span class="co">#NOG274749 / NOG274749      </span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="st">"ENSP00000409316"</span>, <span class="st">"NOG282909"</span>, <span class="co">#NOG282909 / NOG282909 / NOG282909      </span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="st">"ENSP00000480090"</span>, <span class="st">"KOG3599"</span>    <span class="co">#KOG3599 / KOG3272</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing unresolved cases and adding manual assignments</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>gene_cogs <span class="sc">%&lt;&gt;%</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a> dplyr<span class="sc">::</span> <span class="fu">select</span>(<span class="sc">-</span>n) <span class="sc">%&gt;%</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(gene_cogs_resolved)</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="co">#gene_cogs |&gt; </span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="co">#  vroom::vroom_write(file = here("data/gene_cogs.csv"), delim = ",")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="processamento" class="level2" data-number="1.5">
<h2 data-number="1.5" class="anchored" data-anchor-id="processamento"><span class="header-section-number">1.5</span> 3.Processamento</h2>
<p>O objetivo desta etapa é realizar o enraizamento dos genes de interesse utilizando o pacote <strong>GeneBridge</strong>. Para isso, utilizamos as funções <code>newBridge</code>, <code>runBridge</code> e <code>runPermutation</code>, que produzem resultados estatísticos associados aos COGs selecionados em uma árvore filogenética.</p>
<section id="inputs-necessários" class="level3" data-number="1.5.1">
<h3 data-number="1.5.1" class="anchored" data-anchor-id="inputs-necessários"><span class="header-section-number">1.5.1</span> 3.1.Inputs necessários</h3>
<ol type="1">
<li><strong><code>ogdata</code></strong>:
<ul>
<li>Dataset contendo três colunas principais:
<ul>
<li><code>Protein ID</code>: Identificadores das proteínas.</li>
<li><code>COG ID</code>: Clusters de interesse.</li>
<li><code>Specie ID</code>: Identificadores das espécies.</li>
</ul></li>
<li>No exemplo, está sendo utilizado o objeto <code>cogdata</code>.</li>
</ul></li>
<li><strong><code>phyloTree</code></strong>:
<ul>
<li>Árvore filogenética contendo 476 eucariotos, representando a estrutura evolutiva entre as espécies analisadas.</li>
</ul></li>
<li><strong><code>ogids</code></strong>:
<ul>
<li>Lista dos <strong>COGs de interesse</strong>. Esse conjunto é derivado da tabela <code>gene_cogs</code> e inclui os COGs associados às proteínas após o processamento anterior.</li>
</ul></li>
<li><strong><code>refsp</code></strong>:
<ul>
<li>Espécie de referência para o enraizamento. No exemplo, utilizamos <code>9606</code> (humano).</li>
</ul></li>
</ol>
<p>A função <em>getBridge</em> extrai os resultados gerados pelo GeneBridge em formato de tabela. A tabela <em>res</em> contém os resultados estastisticos do enraizamento.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Run GeneBridge</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>cogs_of_interest <span class="ot">&lt;-</span> gene_cogs <span class="sc">%&gt;%</span> <span class="fu">pull</span>(cog_id) <span class="sc">%&gt;%</span> unique</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>ogr <span class="ot">&lt;-</span> <span class="fu">newBridge</span>(<span class="at">ogdata=</span>cogdata, <span class="at">phyloTree=</span>phyloTree, <span class="at">ogids =</span> cogs_of_interest, <span class="at">refsp=</span><span class="st">"9606"</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>ogr <span class="ot">&lt;-</span> <span class="fu">runBridge</span>(ogr, <span class="at">penalty =</span> <span class="dv">2</span>, <span class="at">threshold =</span> <span class="fl">0.5</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>ogr <span class="ot">&lt;-</span> <span class="fu">runPermutation</span>(ogr, <span class="at">nPermutations=</span><span class="dv">1000</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">getBridge</span>(ogr, <span class="at">what=</span><span class="st">"results"</span>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ogr, <span class="at">file =</span> <span class="st">"../data/ogr.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="pós-processamento" class="level2" data-number="1.6">
<h2 data-number="1.6" class="anchored" data-anchor-id="pós-processamento"><span class="header-section-number">1.6</span> 4.Pós-Processamento</h2>
<p>Após realizar o enraizamento com o <strong>GeneBridge</strong>, é necessário ajustar os dados para melhorar a visualização e a interpretação dos resultados. Nessa etapa, adicionamos os nomes dos clados às raízes identificadas, utilizando uma tabela externa que relaciona os identificadores das raízes aos nomes dos clados.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># naming the rooted clades</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>CLADE_NAMES <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/dalmolingroup/neurotransmissionevolution/ctenophora_before_porifera/analysis/geneplast_clade_names.tsv"</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>lca_names <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(CLADE_NAMES)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>groot_df <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"cog_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(cog_id, <span class="at">root =</span> Root) <span class="sc">%&gt;%</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(lca_names) <span class="sc">%&gt;%</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(gene_cogs)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(groot_df)</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="co">#groot_df |&gt; </span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co">#  vroom::vroom_write(file = here("data/groot_df.csv"), delim = ",")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="rede-de-interação-proteína-proteína" class="level3" data-number="1.6.1">
<h3 data-number="1.6.1" class="anchored" data-anchor-id="rede-de-interação-proteína-proteína"><span class="header-section-number">1.6.1</span> 4.1.Rede de Interação Proteína-Proteína</h3>
<p>A construção de uma rede de interação proteína-proteína (PPI) é uma etapa essencial para identificar as relações funcionais entre proteínas. Neste processo, utilizamos a API do <strong>STRINGdb</strong>, um banco de dados que cataloga interações entre proteínas com base em diversas fontes, incluindo ensaios experimentais, co-expressão, e evidências extraídas de publicações científicas.</p>
<p>A API do STRINGdb oferece métodos para: - Obter interações proteicas para uma lista de proteínas. - Selecionar fontes específicas de evidências. - Calcular e combinar escores baseados nas evidências selecionadas.</p>
<p>Mais informações sobre a API podem ser encontradas na <a href="https://string-db.org/help/api/">documentação STRING API</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get proteins interaction</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>string_edgelist <span class="ot">&lt;-</span> <span class="fu">get_network_interaction</span>(groot_df)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Recomputing scores</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>string_edgelist <span class="ot">&lt;-</span> <span class="fu">combine_scores</span>(string_edgelist, </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">evidences =</span> <span class="fu">c</span>(<span class="st">"ascore"</span>, <span class="st">"escore"</span>, <span class="st">"dscore"</span>), </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">confLevel =</span> <span class="fl">0.7</span>)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(string_edgelist) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"stringId_A"</span>, <span class="st">"stringId_B"</span>, <span class="st">"combined_score"</span>)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove o species id</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>string_edgelist<span class="sc">$</span>stringId_A <span class="ot">&lt;-</span> <span class="fu">substring</span>(string_edgelist<span class="sc">$</span>stringId_A, <span class="dv">6</span>, <span class="dv">1000</span>)</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>string_edgelist<span class="sc">$</span>stringId_B <span class="ot">&lt;-</span> <span class="fu">substring</span>(string_edgelist<span class="sc">$</span>stringId_B, <span class="dv">6</span>, <span class="dv">1000</span>)</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co"># How many edgelist proteins are absent in gene_ids? (should return 0)</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  string_edgelist <span class="sc">%$%</span> <span class="fu">c</span>(stringId_A, stringId_B),</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  map_ids <span class="sc">%&gt;%</span> <span class="fu">pull</span>(stringId)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(string_edgelist)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Para a construção do grafo, além das interações entre as proteínas, é necessário que cada nó seja anotado com informações adicionais que serão usadas na análise, como: - Nome da proteína. - Clado onde está enraizado. - Via metabólica em que participa.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Create anotation table</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>nodelist <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">node =</span> <span class="fu">unique</span>(<span class="fu">c</span>(string_edgelist<span class="sc">$</span>stringId_A, string_edgelist<span class="sc">$</span>stringId_B)))</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>merged_paths <span class="ot">&lt;-</span> <span class="fu">merge</span>(nodelist, groot_df, <span class="at">by.x =</span> <span class="st">"node"</span>, <span class="at">by.y =</span> <span class="st">"protein_id"</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>pivotada <span class="ot">&lt;-</span> sensorial_genes <span class="sc">%&gt;%</span> </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_symbol, pathway_name) <span class="sc">%&gt;%</span> </span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">n =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">id_cols =</span> gene_symbol,</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_from =</span> pathway_name,</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_from =</span> n,</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_fn =</span> <span class="fu">list</span>(<span class="at">n =</span> length),</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_fill =</span> <span class="fu">list</span>(<span class="at">n =</span> <span class="dv">0</span>),</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>source_statements <span class="ot">&lt;-</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(pivotada)[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(pivotada)]</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>nodelist <span class="ot">&lt;-</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>  nodelist <span class="sc">%&gt;%</span></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(merged_paths, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"node"</span> <span class="ot">=</span> <span class="st">"node"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(map_ids, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"node"</span> <span class="ot">=</span> <span class="st">"stringId"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(pivotada, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"queryItem"</span> <span class="ot">=</span> <span class="st">"gene_symbol"</span>))</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nodelist)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Além da estrutura do grafo, podemos calcular métricas como o número de conexões (grau) de cada nó.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Network Metrics</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>connected_nodes <span class="ot">&lt;-</span> <span class="fu">rle</span>(<span class="fu">sort</span>(<span class="fu">c</span>(string_edgelist[,<span class="dv">1</span>], string_edgelist[,<span class="dv">2</span>])))</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>connected_nodes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">count=</span>connected_nodes<span class="sc">$</span>lengths, <span class="at">node=</span>connected_nodes<span class="sc">$</span>values)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>connected_nodes <span class="ot">&lt;-</span> <span class="fu">left_join</span>(nodelist, connected_nodes, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"node"</span> <span class="ot">=</span> <span class="st">"node"</span>))</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>connected_nodes <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(connected_nodes, queryItem, root, clade_name, count)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(connected_nodes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#nodelist |&gt; </span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co">#  vroom::vroom_write(file = here("data/nodelist.csv"), delim = ",")</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">#string_edgelist |&gt; </span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  vroom::vroom_write(file = here("data/string_edgelist.csv"), delim = ",")</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#merged_paths |&gt; </span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  vroom::vroom_write(file = here("data/merged_paths.csv"), delim = ",")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="importar-bibliotecas" class="level2" data-number="1.7">
<h2 data-number="1.7" class="anchored" data-anchor-id="importar-bibliotecas"><span class="header-section-number">1.7</span> Importar bibliotecas</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vroom)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(paletteer)</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(easylayout)</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(UpSetR)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tinter)</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="definir-funções" class="level2" data-number="1.8">
<h2 data-number="1.8" class="anchored" data-anchor-id="definir-funções"><span class="header-section-number">1.8</span> Definir funções</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set colors</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>color_mappings <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Olfactory transduction"</span>   <span class="ot">=</span> <span class="st">"#8dd3c7"</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  ,<span class="st">"Taste transduction"</span>      <span class="ot">=</span> <span class="st">"#72874EFF"</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  ,<span class="st">"Phototransduction"</span>       <span class="ot">=</span> <span class="st">"#fb8072"</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>subset_graph_by_root <span class="ot">&lt;-</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(geneplast_result, root_number, graph) {</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>    filtered <span class="ot">&lt;-</span> geneplast_result <span class="sc">%&gt;%</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(root <span class="sc">&gt;=</span> root_number) <span class="sc">%&gt;%</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pull</span>(node)</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">induced_subgraph</span>(graph, <span class="fu">which</span>(<span class="fu">V</span>(graph)<span class="sc">$</span>name <span class="sc">%in%</span> filtered))</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>adjust_color_by_root <span class="ot">&lt;-</span> <span class="cf">function</span>(geneplast_result, root_number, graph) {</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>  filtered <span class="ot">&lt;-</span> geneplast_result <span class="sc">%&gt;%</span></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(root <span class="sc">==</span> root_number) <span class="sc">%&gt;%</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(node)</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">V</span>(graph)<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">V</span>(graph)<span class="sc">$</span>name <span class="sc">%in%</span> filtered, <span class="st">"black"</span>, <span class="st">"gray"</span>)</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(graph)</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Configure graph collors by genes incrementation</span></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>subset_and_adjust_color_by_root <span class="ot">&lt;-</span> <span class="cf">function</span>(geneplast_result, root_number, graph) {</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>  subgraph <span class="ot">&lt;-</span> <span class="fu">subset_graph_by_root</span>(geneplast_result, root_number, graph)</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>  adjusted_graph <span class="ot">&lt;-</span> <span class="fu">adjust_color_by_root</span>(geneplast_result, root_number, subgraph)</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(adjusted_graph)</span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>plot_network <span class="ot">&lt;-</span> <span class="cf">function</span>(graph, title, nodelist, xlims, ylims, <span class="at">legend =</span> <span class="st">"none"</span>) {</span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Generate color map</span></span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>  source_statements <span class="ot">&lt;-</span></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(nodelist)[<span class="dv">10</span><span class="sc">:</span><span class="fu">length</span>(nodelist)]</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>  color_mappings <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Olfactory transduction"</span>   <span class="ot">=</span> <span class="st">"#8dd3c7"</span></span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>    ,<span class="st">"Taste transduction"</span>      <span class="ot">=</span> <span class="st">"#72874EFF"</span></span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>    ,<span class="st">"Phototransduction"</span>       <span class="ot">=</span> <span class="st">"#fb8072"</span></span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>  vertices <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">as_data_frame</span>(graph, <span class="st">"vertices"</span>)</span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>  ggraph<span class="sc">::</span> <span class="fu">ggraph</span>(graph,</span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"manual"</span>,</span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a>                  <span class="at">x =</span> <span class="fu">V</span>(graph)<span class="sc">$</span>x,</span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>                  <span class="at">y =</span> <span class="fu">V</span>(graph)<span class="sc">$</span>y) <span class="sc">+</span></span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>    ggraph<span class="sc">::</span><span class="fu">geom_edge_link0</span>(<span class="at">edge_width =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"#90909020"</span>) <span class="sc">+</span></span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a>    ggraph<span class="sc">::</span><span class="fu">geom_node_point</span>(ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">color =</span> <span class="fu">I</span>(<span class="fu">V</span>(graph)<span class="sc">$</span>color)), <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a>    scatterpie<span class="sc">::</span><span class="fu">geom_scatterpie</span>(</span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a>      <span class="fu">aes</span>(<span class="at">x=</span>x, <span class="at">y=</span>y, <span class="at">r=</span><span class="dv">18</span>),</span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a>      <span class="at">cols =</span> source_statements,</span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> vertices[<span class="fu">rownames</span>(vertices) <span class="sc">%in%</span> <span class="fu">V</span>(graph)<span class="sc">$</span>name[<span class="fu">V</span>(graph)<span class="sc">$</span>color <span class="sc">==</span> <span class="st">"black"</span>],],</span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a>      <span class="at">colour =</span> <span class="cn">NA</span>,</span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a>      <span class="at">pie_scale =</span> <span class="dv">1</span></span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_node_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">ifelse</span>(<span class="fu">V</span>(graph)<span class="sc">$</span>color <span class="sc">==</span> <span class="st">"black"</span>, <span class="fu">V</span>(graph)<span class="sc">$</span>queryItem, <span class="cn">NA</span>)), </span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a>                   <span class="at">nudge_x =</span> <span class="dv">1</span>, <span class="at">nudge_y =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="fl">0.5</span>, <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(<span class="at">values =</span> color_mappings, <span class="at">drop =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_x_continuous</span>(<span class="at">limits =</span> xlims) <span class="sc">+</span></span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> ylims) <span class="sc">+</span></span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb13-67"><a href="#cb13-67" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb13-68"><a href="#cb13-68" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.position =</span> legend,</span>
<span id="cb13-69"><a href="#cb13-69" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.key.size =</span> ggplot2<span class="sc">::</span><span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">'cm'</span>),</span>
<span id="cb13-70"><a href="#cb13-70" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.key.height =</span> ggplot2<span class="sc">::</span><span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">'cm'</span>),</span>
<span id="cb13-71"><a href="#cb13-71" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.key.width =</span> ggplot2<span class="sc">::</span><span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">'cm'</span>),</span>
<span id="cb13-72"><a href="#cb13-72" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.title =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>),</span>
<span id="cb13-73"><a href="#cb13-73" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.text =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>),</span>
<span id="cb13-74"><a href="#cb13-74" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.border =</span> ggplot2<span class="sc">::</span><span class="fu">element_rect</span>(</span>
<span id="cb13-75"><a href="#cb13-75" aria-hidden="true" tabindex="-1"></a>        <span class="at">colour =</span> <span class="st">"#161616"</span>,</span>
<span id="cb13-76"><a href="#cb13-76" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> <span class="cn">NA</span>,</span>
<span id="cb13-77"><a href="#cb13-77" aria-hidden="true" tabindex="-1"></a>        <span class="at">linewidth =</span> <span class="dv">1</span></span>
<span id="cb13-78"><a href="#cb13-78" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb13-79"><a href="#cb13-79" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.title =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">face =</span> <span class="st">"bold"</span>)</span>
<span id="cb13-80"><a href="#cb13-80" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb13-81"><a href="#cb13-81" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">guides</span>(</span>
<span id="cb13-82"><a href="#cb13-82" aria-hidden="true" tabindex="-1"></a>      <span class="at">color =</span> <span class="st">"none"</span>,</span>
<span id="cb13-83"><a href="#cb13-83" aria-hidden="true" tabindex="-1"></a>      <span class="at">fill =</span> <span class="st">"none"</span></span>
<span id="cb13-84"><a href="#cb13-84" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb13-85"><a href="#cb13-85" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">"Source:"</span>, <span class="at">title =</span> title)</span>
<span id="cb13-86"><a href="#cb13-86" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="carregando-tabelas-necessárias" class="level2" data-number="1.9">
<h2 data-number="1.9" class="anchored" data-anchor-id="carregando-tabelas-necessárias"><span class="header-section-number">1.9</span> Carregando tabelas necessárias</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Load data (need to save tables from first qmd)</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>nodelist <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data/nodelist.csv"</span>), <span class="at">delim =</span> <span class="st">","</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>string_edgelist <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data/string_edgelist.csv"</span>), <span class="at">delim =</span> <span class="st">","</span>)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>merged_paths <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data/merged_paths.csv"</span>), <span class="at">delim =</span> <span class="st">","</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="visualização-com-upset-plot" class="level2" data-number="1.10">
<h2 data-number="1.10" class="anchored" data-anchor-id="visualização-com-upset-plot"><span class="header-section-number">1.10</span> 1.Visualização com UpSet Plot</h2>
<p>O <em>UpSet Plot</em> é uma ferramenta útil para visualizar a distribuição e concatenação de genes entre diferentes vias metabólicas. Ele permite identificar como os genes estão compartilhados ou exclusivos entre as categorias analisadas.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">upset</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">as.data.frame</span>(nodelist), </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Olfactory transduction"</span>,</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Taste transduction"</span>,</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Phototransduction"</span>),</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">nsets =</span> <span class="dv">50</span>, <span class="at">nintersects =</span> <span class="cn">NA</span>,</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">sets.bar.color =</span> <span class="fu">c</span>(<span class="st">"#8dd3c7"</span>, <span class="st">"#72874EFF"</span>, <span class="st">"#fb8072"</span>), </span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">mainbar.y.label =</span> <span class="st">"Biological Process </span><span class="sc">\n</span><span class="st">Intersections"</span>,</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">sets.x.label =</span> <span class="st">"Set Size"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="visualização-da-rede-de-interação-proteína-proteína" class="level2" data-number="1.11">
<h2 data-number="1.11" class="anchored" data-anchor-id="visualização-da-rede-de-interação-proteína-proteína"><span class="header-section-number">1.11</span> 2.Visualização da Rede de Interação Proteína-Proteína</h2>
<p>A visualização da rede de interação é essencial para compreender as conexões funcionais entre proteínas. Aqui, utilizamos o pacote <strong>easylayout</strong>, desenvolvido por Danilo Imparato, para gerar um layout eficiente. Este pacote organiza os nós da rede em coordenadas x e y, permitindo uma visualização estruturada e clara. Posteriormente, o grafo será plotado com o <strong>ggraph</strong>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Graph Build</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="co">#graph &lt;-</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  graph_from_data_frame(string_edgelist, directed = FALSE, vertices = nodelist)</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co">#layout &lt;- easylayout::easylayout(graph)</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="co">#V(graph)$x &lt;- layout[, 1]</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co">#V(graph)$y &lt;- layout[, 2]</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="co">#save(graph, file = "../data/graph_layout")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="visualização-da-ancestralidade-de-cada-nó" class="level3" data-number="1.11.1">
<h3 data-number="1.11.1" class="anchored" data-anchor-id="visualização-da-ancestralidade-de-cada-nó"><span class="header-section-number">1.11.1</span> 2.1.Visualização da Ancestralidade de Cada Nó</h3>
<p>A análise da ancestralidade de cada nó na rede fornece uma visão evolutiva sobre as proteínas analisadas. Aqui, utilizamos o <strong>ggraph</strong> para plotar o grafo com as posições previamente salvas pelo <em>easylayout</em>.</p>
<p>Os nós são coloridos de acordo com a distância em relação ao último ancestral comum (LCA) dos clados analisados e o humano (<em>Human-LCA</em>). A tonalidade mais escura indica clados mais antigos em relação ao humano, enquanto tons claros de azul representam clados mais novos, mais próximos do <em>Human-LCA</em>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"../data/graph_layout"</span>)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggraph</span>(graph, <span class="st">"manual"</span>, <span class="at">x =</span> <span class="fu">V</span>(graph)<span class="sc">$</span>x, <span class="at">y =</span> <span class="fu">V</span>(graph)<span class="sc">$</span>y) <span class="sc">+</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_edge_link0</span>(<span class="at">color =</span> <span class="st">"#90909020"</span>) <span class="sc">+</span>  </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_node_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> <span class="sc">-</span>root), <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span>  </span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>() <span class="sc">+</span> </span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"left"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="visualização-da-rede-de-interação-proteína-proteína-em-humano" class="level3" data-number="1.11.2">
<h3 data-number="1.11.2" class="anchored" data-anchor-id="visualização-da-rede-de-interação-proteína-proteína-em-humano"><span class="header-section-number">1.11.2</span> 2.2.Visualização da Rede de Interação Proteína-Proteína em Humano</h3>
<p>Para compreender melhor a relação entre proteínas humanas, plotamos a rede de interação onde os nós representam os genes humanos associados aos seus processos biológicos.</p>
<section id="descrição-dos-elementos-do-gráfico" class="level4" data-number="1.11.2.1">
<h4 data-number="1.11.2.1" class="anchored" data-anchor-id="descrição-dos-elementos-do-gráfico"><span class="header-section-number">1.11.2.1</span> Descrição dos elementos do gráfico:</h4>
<ol type="1">
<li><strong>Nós (Círculos):</strong> As cores dos nós são divididas de acordo com os processos biológicos atribuídos a cada gene. O uso de diagramas de pizza permite a visualização de genes que participam de múltiplos processos.</li>
<li><strong>Arestas (Linhas):</strong> Representam as interações proteicas baseadas em dados do STRINGdb.</li>
<li><strong>Rótulos dos genes:</strong> Cada nó está anotado com o símbolo do gene correspondente, posicionado estrategicamente para facilitar a leitura.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Plotting Human PPI Network</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="co">#ppi_labaled &lt;-</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>ggraph<span class="sc">::</span><span class="fu">ggraph</span>(graph,</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>               <span class="st">"manual"</span>,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> <span class="fu">V</span>(graph)<span class="sc">$</span>x,</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> <span class="fu">V</span>(graph)<span class="sc">$</span>y) <span class="sc">+</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  ggraph<span class="sc">::</span> <span class="fu">geom_edge_link0</span>(<span class="at">edge_width =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"#90909020"</span>) <span class="sc">+</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  scatterpie<span class="sc">::</span><span class="fu">geom_scatterpie</span>(</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">colnames</span>(nodelist[<span class="dv">10</span><span class="sc">:</span><span class="dv">12</span>]),</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> igraph<span class="sc">::</span><span class="fu">as_data_frame</span>(graph, <span class="st">"vertices"</span>),</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="cn">NA</span>,</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">pie_scale =</span> <span class="fl">0.40</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_node_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> nodelist<span class="sc">$</span>queryItem), <span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">nudge_x =</span> <span class="fl">0.8</span>, <span class="at">nudge_y =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(<span class="at">values =</span> color_mappings, <span class="at">drop =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ppi &lt;- </span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  ggraph<span class="sc">::</span><span class="fu">ggraph</span>(graph,</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>               <span class="st">"manual"</span>,</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> <span class="fu">V</span>(graph)<span class="sc">$</span>x,</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> <span class="fu">V</span>(graph)<span class="sc">$</span>y) <span class="sc">+</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  ggraph<span class="sc">::</span> <span class="fu">geom_edge_link0</span>(<span class="at">edge_width =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"#90909020"</span>) <span class="sc">+</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  scatterpie<span class="sc">::</span><span class="fu">geom_scatterpie</span>(</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">colnames</span>(nodelist[<span class="dv">10</span><span class="sc">:</span><span class="dv">12</span>]),</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> igraph<span class="sc">::</span><span class="fu">as_data_frame</span>(graph, <span class="st">"vertices"</span>),</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="cn">NA</span>,</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">pie_scale =</span> <span class="fl">0.40</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(<span class="at">values =</span> color_mappings, <span class="at">drop =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-35-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="visualização-da-rede-de-interação-proteína-proteína-em-cada-clado" class="level3" data-number="1.11.3">
<h3 data-number="1.11.3" class="anchored" data-anchor-id="visualização-da-rede-de-interação-proteína-proteína-em-cada-clado"><span class="header-section-number">1.11.3</span> 2.3.Visualização da Rede de Interação Proteína-Proteína em Cada Clado</h3>
<p>Nesta seção, visualizamos os genes que estão estatisticamente enraizados em cada clado. A disposição dos genes permite observar o incremento dos genes ortólogos em função da complexidade e antiguidade do sistema biológico.</p>
<section id="características-da-visualização" class="level4" data-number="1.11.3.1">
<h4 data-number="1.11.3.1" class="anchored" data-anchor-id="características-da-visualização"><span class="header-section-number">1.11.3.1</span> Características da visualização:</h4>
<ol type="1">
<li><strong>Evolução dos grafos:</strong> Os grafos são organizados da esquerda para a direita e de cima para baixo, permitindo analisar a progressão evolutiva.</li>
<li><strong>Coloração dos nós:</strong> A cor dos nós indica o nível de ancestralidade, como previamente destacado, onde tons mais escuros representam clados mais antigos e tons mais claros indicam proximidade evolutiva com humanos.</li>
<li><strong>Organismos de interesse:</strong> Além de visualizar todos os clados, é possível gerar gráficos focados apenas em determinados grupos, como <em>Metamonada</em>, <em>Choanoflagellata</em>, <em>Cephalochordata</em> e <em>Amphibia</em>.</li>
</ol>
<p>Com estas visualizações, é possível identificar padrões de evolução dos genes em diferentes clados e realizar comparações detalhadas com organismos de interesse específico.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>geneplast_roots <span class="ot">&lt;-</span> merged_paths[<span class="fu">order</span>(merged_paths<span class="sc">$</span>root), ]</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>buffer <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">50</span>, <span class="dv">50</span>)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>xlims <span class="ot">&lt;-</span> <span class="fu">ceiling</span>(<span class="fu">range</span>(<span class="fu">V</span>(graph)<span class="sc">$</span>x)) <span class="sc">+</span> buffer</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>ylims <span class="ot">&lt;-</span> <span class="fu">ceiling</span>(<span class="fu">range</span>(<span class="fu">V</span>(graph)<span class="sc">$</span>y)) <span class="sc">+</span> buffer</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>roots <span class="ot">&lt;-</span> <span class="fu">unique</span>(geneplast_roots<span class="sc">$</span>root) <span class="sc">%&gt;%</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(<span class="fu">unique</span>(geneplast_roots<span class="sc">$</span>clade_name))</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset graphs by LCAs</span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>subsets <span class="ot">&lt;-</span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(roots, <span class="sc">~</span> <span class="fu">subset_and_adjust_color_by_root</span>(geneplast_roots, .x, graph))</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot titles</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>titles <span class="ot">&lt;-</span> <span class="fu">names</span>(roots)</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>    subsets,</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>    titles,</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>    plot_network,</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">nodelist =</span> nodelist,</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlims =</span> xlims,</span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylims =</span> ylims,</span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend =</span> <span class="st">"right"</span></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">discard</span>(is.null)</span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a><span class="co">#net_all_roots &lt;-</span></span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>patchwork<span class="sc">::</span><span class="fu">wrap_plots</span>(</span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rev</span>(plots),</span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">nrow =</span> <span class="dv">4</span>,</span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">4</span></span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(file = "../data/network_rooting.svg", plot=net_all_roots, width=10, height=8)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>patchwork<span class="sc">::</span><span class="fu">wrap_plots</span>(</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  plots<span class="sc">$</span>Metamonada, plots<span class="sc">$</span>Choanoflagellata, plots<span class="sc">$</span>Cephalochordata, plots<span class="sc">$</span>Amphibia,</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">4</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="importando-pacotes-1" class="level2" data-number="1.12">
<h2 data-number="1.12" class="anchored" data-anchor-id="importando-pacotes-1"><span class="header-section-number">1.12</span> Importando Pacotes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hrbrthemes)</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tinter)</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="definindo-funções" class="level2" data-number="1.13">
<h2 data-number="1.13" class="anchored" data-anchor-id="definindo-funções"><span class="header-section-number">1.13</span> Definindo funções</h2>
<p>Essas funções possuem a finalidade de organizar a anotação dos vértices de nossa nodelist e calcular o cumulativo de genes por clado e por processo biológico</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>calculate_cumulative_genes <span class="ot">&lt;-</span> <span class="cf">function</span>(nodelist) {</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Obter todas as categorias possíveis de clade_name</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  all_clades <span class="ot">&lt;-</span> node_annotation <span class="sc">%&gt;%</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(root)) <span class="sc">%&gt;%</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span> <span class="fu">select</span>(clade_name) <span class="sc">%&gt;%</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>()</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Definir as colunas de interesse</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  process_columns <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"queryItem"</span>, <span class="st">"root"</span>, <span class="st">"clade_name"</span>, </span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"Olfactory transduction"</span>, </span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"Taste transduction"</span>,           </span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"Phototransduction"</span>)</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calcular o cumulativo agrupando por clade_name</span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>  cumulative_genes <span class="ot">&lt;-</span> nodelist <span class="sc">%&gt;%</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(root)) <span class="sc">%&gt;%</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(process_columns)) <span class="sc">%&gt;%</span></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(clade_name, root) <span class="sc">%&gt;%</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">count_genes =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(root)) <span class="sc">%&gt;%</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">cumulative_sum =</span> <span class="fu">cumsum</span>(count_genes)) <span class="sc">%&gt;%</span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">right_join</span>(all_clades, <span class="at">by =</span> <span class="st">"clade_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fill</span>(cumulative_sum, <span class="at">.direction =</span> <span class="st">"down"</span>)</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(cumulative_genes)</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>calculate_cumulative_bp <span class="ot">&lt;-</span> <span class="cf">function</span>(nodelist) {</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Obter todas as categorias possíveis de clade_name</span></span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>  all_clades <span class="ot">&lt;-</span> node_annotation <span class="sc">%&gt;%</span></span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(root)) <span class="sc">%&gt;%</span></span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span> <span class="fu">select</span>(clade_name) <span class="sc">%&gt;%</span></span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>()</span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>   <span class="co"># Definir as colunas de interesse</span></span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a>  process_columns <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"queryItem"</span>, <span class="st">"root"</span>, <span class="st">"clade_name"</span>, </span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"Olfactory transduction"</span>, </span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"Taste transduction"</span>,           </span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"Phototransduction"</span>)</span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calcular a soma cumulativa para cada processo biológico</span></span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a>  cumulative_bp <span class="ot">&lt;-</span> nodelist <span class="sc">%&gt;%</span></span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(process_columns)) <span class="sc">%&gt;%</span></span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(root, queryItem, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">all_of</span>(process_columns[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)]), <span class="sc">~</span> <span class="fu">as.numeric</span>(.))) <span class="sc">%&gt;%</span></span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(root, clade_name) <span class="sc">%&gt;%</span></span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">all_of</span>(process_columns[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)]), </span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a>                     <span class="sc">~</span> <span class="fu">sum</span>(. , <span class="at">na.rm =</span> <span class="cn">TRUE</span>)),</span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a>              <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(root)) <span class="sc">%&gt;%</span></span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">all_of</span>(process_columns[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)]), <span class="sc">~</span> <span class="fu">cumsum</span>(.))) <span class="sc">%&gt;%</span></span>
<span id="cb24-54"><a href="#cb24-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">right_join</span>(all_clades, <span class="at">by =</span> <span class="st">"clade_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-55"><a href="#cb24-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fill</span>(<span class="fu">everything</span>(), <span class="at">.direction =</span> <span class="st">"down"</span>)</span>
<span id="cb24-56"><a href="#cb24-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-57"><a href="#cb24-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(cumulative_bp)</span>
<span id="cb24-58"><a href="#cb24-58" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="definindo-parâmetros-estéticos-para-o-ggplot" class="level2" data-number="1.14">
<h2 data-number="1.14" class="anchored" data-anchor-id="definindo-parâmetros-estéticos-para-o-ggplot"><span class="header-section-number">1.14</span> Definindo parâmetros estéticos para o ggplot</h2>
<p>Atribuindo cores padrões para as vias metabólicas da análise e definindo os demais padrões estéticos para plotagens</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting colors and labels</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>annotation_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Olfactory transduction"</span>   <span class="ot">=</span> <span class="st">"#8dd3c7"</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  ,<span class="st">"Taste transduction"</span>      <span class="ot">=</span> <span class="st">"#72874EFF"</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  ,<span class="st">"Phototransduction"</span>       <span class="ot">=</span> <span class="st">"#fb8072"</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>annotation_labels <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Olfactory transduction"</span>   <span class="ot">=</span> <span class="st">"Olfactory transduction"</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>  ,<span class="st">"Taste transduction"</span>      <span class="ot">=</span> <span class="st">"Taste transduction"</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>  ,<span class="st">"Phototransduction"</span>       <span class="ot">=</span> <span class="st">"Phototransduction"</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a><span class="co"># This vertical line indicates the first metazoan (Amphimedon queenslandica / Ctenophora)</span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>choanoflagellata_line <span class="ot">&lt;-</span> <span class="fu">geom_vline</span>(</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">xintercept =</span> <span class="st">"Sphaeroforma arctica"</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">color      =</span> <span class="st">"#FF0000"</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">linetype   =</span> <span class="st">"11"</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">alpha      =</span> <span class="dv">1</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">linewidth  =</span> <span class="fl">0.25</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting</span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>theme_main <span class="ot">&lt;-</span> <span class="fu">theme</span>(</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">panel.spacing      =</span> <span class="fu">unit</span>(<span class="fl">2.5</span>, <span class="st">"pt"</span>)</span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">strip.background   =</span> <span class="fu">element_blank</span>()</span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">panel.grid.major.y =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> <span class="fl">0.25</span>, <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">color =</span> <span class="st">"#E0E0E0"</span>)</span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">strip.text.x       =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"#757575"</span>)</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">strip.text.y       =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">angle =</span> <span class="dv">0</span>, <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"#757575"</span>)</span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.title         =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>, <span class="at">color =</span> <span class="st">"#424242"</span>)</span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.ticks.x       =</span> <span class="fu">element_blank</span>()</span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.text.x        =</span> <span class="fu">element_blank</span>()</span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.text.y        =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fl">5.5</span>)</span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">legend.position    =</span> <span class="st">"none"</span></span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a>theme_supplementary <span class="ot">&lt;-</span> <span class="fu">theme</span>(</span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a>  <span class="at">panel.grid.major.x =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"#E0E0E0"</span>, <span class="at">linewidth =</span> <span class="fl">0.25</span>, <span class="at">linetype =</span> <span class="st">"dotted"</span>)</span>
<span id="cb25-40"><a href="#cb25-40" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">panel.grid.major.y =</span> <span class="fu">element_blank</span>()</span>
<span id="cb25-41"><a href="#cb25-41" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">strip.text.y       =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>, <span class="at">angle =</span> <span class="dv">0</span>, <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"#757575"</span>)</span>
<span id="cb25-42"><a href="#cb25-42" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">strip.text.x       =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"#757575"</span>)</span>
<span id="cb25-43"><a href="#cb25-43" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.title         =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">color =</span> <span class="st">"#424242"</span>)</span>
<span id="cb25-44"><a href="#cb25-44" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.ticks         =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"grey20"</span>)</span>
<span id="cb25-45"><a href="#cb25-45" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.text.y        =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>, <span class="at">angle =</span> <span class="dv">0</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"#757575"</span>)</span>
<span id="cb25-46"><a href="#cb25-46" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.text.x        =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>)</span>
<span id="cb25-47"><a href="#cb25-47" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-48"><a href="#cb25-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-49"><a href="#cb25-49" aria-hidden="true" tabindex="-1"></a>theme_average <span class="ot">&lt;-</span> <span class="fu">theme</span>(</span>
<span id="cb25-50"><a href="#cb25-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">panel.spacing      =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"pt"</span>)</span>
<span id="cb25-51"><a href="#cb25-51" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.title         =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"#424242"</span>)</span>
<span id="cb25-52"><a href="#cb25-52" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.text          =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"#757575"</span>)</span>
<span id="cb25-53"><a href="#cb25-53" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.text.x        =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>, <span class="at">angle =</span> <span class="sc">-</span><span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">0</span>, <span class="at">hjust =</span> <span class="dv">0</span>)</span>
<span id="cb25-54"><a href="#cb25-54" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.text.y        =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">5</span>)</span>
<span id="cb25-55"><a href="#cb25-55" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">strip.background   =</span> <span class="fu">element_blank</span>()</span>
<span id="cb25-56"><a href="#cb25-56" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">strip.text         =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"#757575"</span>)</span>
<span id="cb25-57"><a href="#cb25-57" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">strip.text.y       =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb25-58"><a href="#cb25-58" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-59"><a href="#cb25-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-60"><a href="#cb25-60" aria-hidden="true" tabindex="-1"></a>theme_big <span class="ot">&lt;-</span> <span class="fu">theme</span>(</span>
<span id="cb25-61"><a href="#cb25-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">panel.spacing      =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"pt"</span>)</span>
<span id="cb25-62"><a href="#cb25-62" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">panel.grid.major.x =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> <span class="fl">0.1</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>)</span>
<span id="cb25-63"><a href="#cb25-63" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">panel.grid.major.y =</span> <span class="fu">element_blank</span>()</span>
<span id="cb25-64"><a href="#cb25-64" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">strip.background   =</span> <span class="fu">element_blank</span>()</span>
<span id="cb25-65"><a href="#cb25-65" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">strip.text.x       =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">vjust =</span> <span class="dv">0</span>)</span>
<span id="cb25-66"><a href="#cb25-66" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">strip.text.y       =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">0</span>, <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb25-67"><a href="#cb25-67" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.text.x        =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="dv">0</span>, <span class="at">hjust =</span> <span class="dv">0</span>)</span>
<span id="cb25-68"><a href="#cb25-68" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.text.y        =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fl">4.5</span>)</span>
<span id="cb25-69"><a href="#cb25-69" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">axis.ticks         =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fl">0.1</span>)</span>
<span id="cb25-70"><a href="#cb25-70" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-71"><a href="#cb25-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-72"><a href="#cb25-72" aria-hidden="true" tabindex="-1"></a>tick_function <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb25-73"><a href="#cb25-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">seq</span>(x[<span class="dv">2</span>], <span class="dv">0</span>, <span class="at">length.out =</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="sc">-</span><span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">tail</span>(<span class="sc">-</span><span class="dv">1</span>) <span class="sc">%&gt;%</span> { <span class="fu">ceiling</span>(.<span class="sc">/</span><span class="dv">5</span>)<span class="sc">*</span><span class="dv">5</span> }</span>
<span id="cb25-74"><a href="#cb25-74" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="carregar-tabelas-necessárias" class="level2" data-number="1.15">
<h2 data-number="1.15" class="anchored" data-anchor-id="carregar-tabelas-necessárias"><span class="header-section-number">1.15</span> Carregar tabelas necessárias</h2>
<p>Para prosseguir com a análise, é necessário carregar uma tabela contendo informações do <em>taxid</em> de cada espécie. Essas informações serão posteriormente utilizadas para mapear os <em>taxid</em> das espécies com os <em>taxid</em> de cada COG.</p>
<section id="descrição-da-tabela" class="level3" data-number="1.15.1">
<h3 data-number="1.15.1" class="anchored" data-anchor-id="descrição-da-tabela"><span class="header-section-number">1.15.1</span> Descrição da Tabela</h3>
<ul>
<li><strong>Nome do Arquivo</strong>: <code>string_eukaryotes.rda</code><br>
</li>
<li><strong>Conteúdo</strong>: Informações taxonômicas de espécies eucarióticas.</li>
</ul>
<p>A tabela carrega os seguintes campos principais:<br>
- <strong>TaxID</strong> (<em>taxonomic identifier</em>): Um identificador único associado a cada espécie. - Nome da espécie</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Query Phylotree and OG data</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>ah <span class="ot">&lt;-</span> <span class="fu">AnnotationHub</span>()</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">query</span>(ah, <span class="st">"geneplast"</span>)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(meta[[<span class="st">"AH83116"</span>]])</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Todo: salvar tabelas dos qmd anteriores</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>nodelist <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data/nodelist.csv"</span>), <span class="at">delim =</span> <span class="st">","</span>)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>gene_cogs <span class="ot">&lt;-</span>  vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data/gene_cogs.csv"</span>), <span class="at">delim =</span> <span class="st">","</span>)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>sensorial_genes <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/sensorial_genes.csv"</span>)</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>map_ids <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="st">"../data/map_ids.csv"</span>)</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>groot_df <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="st">"../data/groot_df.csv"</span>)</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>ogr <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/ogr.RData"</span>)</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"../data/string_eukaryotes.rda"</span>)</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(string_eukaryotes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="visualização-padrão-de-enraizamento" class="level2" data-number="1.16">
<h2 data-number="1.16" class="anchored" data-anchor-id="visualização-padrão-de-enraizamento"><span class="header-section-number">1.16</span> Visualização: Padrão de Enraizamento</h2>
<p>Essa visualização permite analisar o padrão cumulativo de surgimento de genes ao longo do tempo evolutivo. Além disso, também mostra o crescimento dos processos biológicos separados por cada via metabólica.</p>
<section id="características-dos-gráficos" class="level3" data-number="1.16.1">
<h3 data-number="1.16.1" class="anchored" data-anchor-id="características-dos-gráficos"><span class="header-section-number">1.16.1</span> Características dos Gráficos</h3>
<ol type="1">
<li><strong>Gráfico de Barras (Padrão Cumulativo)</strong>:
<ul>
<li>Representa o somatório cumulativo de genes associados a cada clado.</li>
<li>Cada barra exibe o total acumulado de genes em determinado clado, permitindo identificar o padrão de diversificação.</li>
</ul></li>
<li><strong>Gráfico Combinado (Barras e Linhas)</strong>:
<ul>
<li>As barras indicam o somatório cumulativo de genes por clado.</li>
<li>As linhas representam o crescimento de diferentes processos biológicos (<em>Process</em>), separados por vias metabólicas, permitindo a comparação entre o surgimento cumulativo de genes e o desenvolvimento de funções biológicas.</li>
</ul></li>
</ol>
</section>
<section id="interpretação" class="level3" data-number="1.16.2">
<h3 data-number="1.16.2" class="anchored" data-anchor-id="interpretação"><span class="header-section-number">1.16.2</span> Interpretação</h3>
<ul>
<li><strong>Padrão Cumulativo</strong>: O gráfico de barras mostra como os genes foram surgindo ao longo do tempo, cumulativamente. Essa visão ajuda a identificar momentos de maior diversificação genética.</li>
<li><strong>Comparação por Processos Biológicos</strong>: O gráfico combinado permite observar quais processos biológicos se destacam em diferentes momentos da evolução e como eles estão relacionados ao surgimento de novos genes.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Mapping roots and proteins info</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>node_annotation <span class="ot">&lt;-</span> nodelist <span class="sc">%&gt;%</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(gene_cogs, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"node"</span> <span class="ot">=</span> <span class="st">"protein_id"</span>, <span class="st">"cog_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(sensorial_genes, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"queryItem"</span> <span class="ot">=</span> <span class="st">"gene_symbol"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(queryItem, cog_id, pathway_name, root, clade_name)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>cumulative_genes <span class="ot">&lt;-</span> <span class="fu">calculate_cumulative_genes</span>(nodelist) </span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>cumulative_bp <span class="ot">&lt;-</span> <span class="fu">calculate_cumulative_bp</span>(nodelist)</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>cumulative_data <span class="ot">&lt;-</span> <span class="fu">left_join</span>(cumulative_genes, cumulative_bp)</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>long_data <span class="ot">&lt;-</span> cumulative_data <span class="sc">%&gt;%</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">5</span><span class="sc">:</span><span class="dv">7</span>, </span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Process"</span>, </span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>)</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a><span class="co">#a &lt;-</span></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Gráfico de barras para cumulative_sum</span></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">data =</span> cumulative_data, </span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>           <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(clade_name, <span class="at">levels =</span> clade_name), <span class="at">y =</span> cumulative_sum), </span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>           <span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"darkgray"</span>, <span class="at">colour =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> cumulative_data, </span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(clade_name, <span class="at">levels =</span> clade_name), <span class="at">y =</span> cumulative_sum, <span class="at">label =</span> cumulative_sum), </span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>            <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">color =</span> <span class="st">"darkgray"</span>) <span class="sc">+</span></span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> annotation_colors) <span class="sc">+</span></span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Clade Name"</span>, <span class="at">y =</span> <span class="st">"Cumulative Sum"</span>, </span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Cumulative Sum and Biological Processes"</span>,</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">"Cumulative Sum"</span>,</span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Biological Processes"</span>) <span class="sc">+</span></span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>  theme_main <span class="sc">+</span></span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-53-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co">#b &lt;- </span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Gráfico de barras para cumulative_sum</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">data =</span> cumulative_data, </span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>           <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(<span class="sc">-</span>root), <span class="at">y =</span> cumulative_sum), </span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"darkgray"</span>, <span class="at">colour =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> cumulative_data, </span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(<span class="sc">-</span>root), <span class="at">y =</span> cumulative_sum, <span class="at">label =</span> cumulative_sum), </span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">color =</span> <span class="st">"darkgray"</span>) <span class="sc">+</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Gráfico de linhas para os processos biológicos</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> long_data, </span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(<span class="sc">-</span>root), <span class="at">y =</span> Value, <span class="at">color =</span> Process, <span class="at">group =</span> Process), </span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Usar a paleta de cores definida</span></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> annotation_colors) <span class="sc">+</span></span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Clade Name"</span>, <span class="at">y =</span> <span class="st">"Cumulative Sum"</span>, </span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Cumulative Sum and Biological Processes"</span>,</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">"Cumulative Sum"</span>,</span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Biological Processes"</span>) <span class="sc">+</span></span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>  theme_main <span class="sc">+</span></span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-53-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(file = "", plot=a, width=15, height=8)</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(file = "", plot=b, width=15, height=8)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="cálculo-da-abundância-de-cogs-por-função-nas-espécies" class="level2" data-number="1.17">
<h2 data-number="1.17" class="anchored" data-anchor-id="cálculo-da-abundância-de-cogs-por-função-nas-espécies"><span class="header-section-number">1.17</span> Cálculo da Abundância de COGs por Função nas Espécies</h2>
<p>Esta análise parte da premissa de que COGs similares compartilham a mesma função biológica. Assim, é possível calcular a abundância média das funções associadas aos COGs em cada espécie, utilizando a quantidade de COGs presentes na espécie identificada pelo <em>TaxID</em>.</p>
<section id="etapas-do-cálculo" class="level3" data-number="1.17.1">
<h3 data-number="1.17.1" class="anchored" data-anchor-id="etapas-do-cálculo"><span class="header-section-number">1.17.1</span> Etapas do Cálculo</h3>
<ol type="1">
<li><strong>Mapeamento das Espécies</strong><br>
Cada espécie foi mapeada para seu respectivo clado (nível hierárquico) com base nas informações do <em>TaxID</em>.
<ul>
<li>Foi utilizada a tabela <code>ogr@spbranches</code> para associar espécies (<em>ssp_id</em>) com seus <em>branches</em> (<em>lca</em>).</li>
<li>Os dados foram organizados para incluir a ordem dos <em>TaxIDs</em>.</li>
</ul></li>
<li><strong>Anotação dos COGs</strong>
<ul>
<li>A relação entre proteínas e genes foi enriquecida com informações funcionais, como identificadores de COGs e nomes de vias metabólicas (<em>pathway_name</em>).</li>
<li>Duplicatas foram removidas para garantir que apenas informações únicas fossem consideradas.</li>
</ul></li>
<li><strong>Integração de Informações das Espécies</strong>
<ul>
<li>Foi criado um mapeamento entre espécies (<em>TaxID</em>) e seus respectivos clados para adicionar as informações taxonômicas relevantes.</li>
<li>As espécies foram ordenadas com base em seus clados, para facilitar a análise hierárquica.</li>
</ul></li>
<li><strong>Cálculo da Abundância Média por Função</strong>
<ul>
<li>Para cada espécie e função metabólica (<em>pathway_name</em>), foi calculada a abundância média dos COGs.</li>
<li>Informações adicionais, como nomes das espécies (<em>ncbi_name</em>) e clados (<em>clade_name</em>), foram integradas ao resultado.</li>
</ul></li>
<li><strong>Ajuste da Abundância (Capping)</strong>
<ul>
<li>Para evitar valores extremos, a abundância média foi ajustada:
<ul>
<li>Valores acima de um limite (média + 3 desvios padrão) foram truncados para 100.</li>
<li>Os valores ajustados foram calculados separadamente por função metabólica.</li>
</ul></li>
</ul></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>lca_spp <span class="ot">&lt;-</span> ogr<span class="sc">@</span>spbranches <span class="sc">%&gt;%</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"taxid"</span> <span class="ot">=</span> ssp_id, <span class="st">"species"</span> <span class="ot">=</span> ssp_name, <span class="st">"lca"</span> <span class="ot">=</span> <span class="st">"branch"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">taxid_order =</span> <span class="fu">row_number</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(lca, taxid, taxid_order)</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>clade_taxids <span class="ot">&lt;-</span> lca_spp</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>clade_names <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="st">"https://raw.githubusercontent.com/dalmolingroup/neurotransmissionevolution/ctenophora_before_porifera/analysis/geneplast_clade_names.tsv"</span>)</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>cog_annotation <span class="ot">&lt;-</span> map_ids <span class="sc">%&gt;%</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(groot_df, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"stringId"</span> <span class="ot">=</span> <span class="st">"protein_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sensorial_genes, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"queryItem"</span> <span class="ot">=</span> <span class="st">"gene_symbol"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(queryItem, cog_id, pathway_name) <span class="sc">%&gt;%</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(cog_id, pathway_name) <span class="sc">%&gt;%</span></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>cog_abundance_by_taxid <span class="ot">&lt;-</span> cogdata <span class="sc">%&gt;%</span></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cog_id <span class="sc">%in%</span> nodelist[[<span class="st">"cog_id"</span>]]) <span class="sc">%&gt;%</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(ssp_id, cog_id, <span class="at">name =</span> <span class="st">"abundance"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(cog_annotation, <span class="at">by =</span> <span class="st">"cog_id"</span>)</span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Mapping species to clade info</span></span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>ordered_species <span class="ot">&lt;-</span> string_eukaryotes <span class="sc">%&gt;%</span></span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(taxid, ncbi_name) <span class="sc">%&gt;%</span></span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(clade_taxids, <span class="at">by =</span> <span class="st">"taxid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(clade_names, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"lca"</span> <span class="ot">=</span> <span class="st">"root"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(lca)) <span class="sc">%&gt;%</span></span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>taxid_order)</span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>avg_abundance_by_function <span class="ot">&lt;-</span> cog_abundance_by_taxid <span class="sc">%&gt;%</span></span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ssp_id, pathway_name) <span class="sc">%&gt;%</span></span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">avg_abundance =</span> <span class="fu">mean</span>(abundance)) <span class="sc">%&gt;%</span></span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Adding species and clade info</span></span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(ordered_species <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> <span class="fu">as.double</span>(taxid)), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ssp_id"</span> <span class="ot">=</span> <span class="st">"taxid"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(lca)) <span class="sc">%&gt;%</span></span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ncbi_name =</span> <span class="fu">factor</span>(ncbi_name, <span class="at">levels =</span> <span class="fu">unique</span>(ncbi_name)),</span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">clade_name =</span> <span class="fu">factor</span>(clade_name, <span class="at">levels =</span> <span class="fu">unique</span>(clade_name))) <span class="sc">%&gt;%</span></span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-43"><a href="#cb30-43" aria-hidden="true" tabindex="-1"></a>capped_abundance_by_function <span class="ot">&lt;-</span> avg_abundance_by_function <span class="sc">%&gt;%</span></span>
<span id="cb30-44"><a href="#cb30-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mutate(capped_abundance = ifelse(abundance &gt;= 100, 100, abundance)) %&gt;%</span></span>
<span id="cb30-45"><a href="#cb30-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(pathway_name) <span class="sc">%&gt;%</span></span>
<span id="cb30-46"><a href="#cb30-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb30-47"><a href="#cb30-47" aria-hidden="true" tabindex="-1"></a>    <span class="co"># max_abundance = max(abundance[lca &lt;= 29])</span></span>
<span id="cb30-48"><a href="#cb30-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">max_abundance =</span> avg_abundance[lca <span class="sc">&lt;=</span> <span class="dv">29</span>] <span class="sc">%&gt;%</span> { <span class="fu">mean</span>(.) <span class="sc">+</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span>(.) }</span>
<span id="cb30-49"><a href="#cb30-49" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">abundance     =</span> <span class="fu">ifelse</span>(avg_abundance <span class="sc">&gt;=</span> max_abundance, <span class="fu">pmin</span>(max_abundance, <span class="dv">100</span>), <span class="fu">pmin</span>(avg_abundance, <span class="dv">100</span>)))</span>
<span id="cb30-50"><a href="#cb30-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-51"><a href="#cb30-51" aria-hidden="true" tabindex="-1"></a><span class="co"># List of signatures</span></span>
<span id="cb30-52"><a href="#cb30-52" aria-hidden="true" tabindex="-1"></a>signatures <span class="ot">&lt;-</span> <span class="fu">unique</span>(node_annotation<span class="sc">$</span>pathway_name)</span>
<span id="cb30-53"><a href="#cb30-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-54"><a href="#cb30-54" aria-hidden="true" tabindex="-1"></a>roots_seq <span class="ot">&lt;-</span> node_annotation <span class="sc">%&gt;%</span></span>
<span id="cb30-55"><a href="#cb30-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(root)) <span class="sc">%&gt;%</span></span>
<span id="cb30-56"><a href="#cb30-56" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span> <span class="fu">select</span>(root, clade_name) <span class="sc">%&gt;%</span></span>
<span id="cb30-57"><a href="#cb30-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb30-58"><a href="#cb30-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-59"><a href="#cb30-59" aria-hidden="true" tabindex="-1"></a>roots_seq<span class="sc">$</span>clade_name <span class="ot">&lt;-</span> <span class="fu">factor</span>(roots_seq<span class="sc">$</span>clade_name, <span class="at">levels =</span> roots_seq<span class="sc">$</span>clade_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="visualização" class="level3" data-number="1.17.2">
<h3 data-number="1.17.2" class="anchored" data-anchor-id="visualização"><span class="header-section-number">1.17.2</span> Visualização</h3>
<p>Nos gráficos a seguir, é possível observar a abundância média dos COGs em cada clado. Cada barra representa uma espécie dentro do respectivo clado, enquanto as cores indicam as diferentes funções metabólicas associadas aos grupos ortólogos.</p>
<section id="gráfico-de-abundância-média-de-cogs-por-clado" class="level4" data-number="1.17.2.1">
<h4 data-number="1.17.2.1" class="anchored" data-anchor-id="gráfico-de-abundância-média-de-cogs-por-clado"><span class="header-section-number">1.17.2.1</span> Gráfico de Abundância Média de COGs por Clado</h4>
<ul>
<li>O eixo <strong>X</strong> representa as espécies.</li>
<li>O eixo <strong>Y</strong> indica a abundância média de proteínas em grupos ortólogos.</li>
<li>As barras são coloridas de acordo com a função metabólica (<em>pathway_name</em>).</li>
</ul>
</section>
<section id="gráfico-com-abundância-ajustada-capped" class="level4" data-number="1.17.2.2">
<h4 data-number="1.17.2.2" class="anchored" data-anchor-id="gráfico-com-abundância-ajustada-capped"><span class="header-section-number">1.17.2.2</span> Gráfico com Abundância Ajustada (Capped)</h4>
<ul>
<li>Para evitar distorções causadas por valores extremos, a abundância foi ajustada a um limite máximo (média + 3 desvios padrão).</li>
<li>Este gráfico fornece uma visualização mais equilibrada, especialmente para funções metabólicas com valores muito elevados.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting by species</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(avg_abundance_by_function) <span class="sc">+</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Geoms  ----------------</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>choanoflagellata_line <span class="sc">+</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> ncbi_name, <span class="at">y =</span> avg_abundance, <span class="at">fill =</span> pathway_name, <span class="at">color =</span> <span class="fu">after_scale</span>(<span class="fu">darken</span>(fill, <span class="fl">0.1</span>)))</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">stat =</span> <span class="st">"identity"</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Labels  ---------------</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a> <span class="fu">xlab</span>(<span class="st">"Espécies"</span>) <span class="sc">+</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Abundância média de proteínas em grupos ortólogos"</span>) <span class="sc">+</span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylab("Average protein abundance in orthologous groups") +</span></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Scales ----------------</span></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a><span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> tick_function, <span class="at">minor_breaks =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> annotation_colors <span class="sc">%&gt;%</span> <span class="fu">darken</span>(<span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Styling ---------------</span></span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a><span class="fu">facet_grid</span>(</span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>  pathway_name <span class="sc">~</span> clade_name</span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">scales   =</span> <span class="st">"free"</span></span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">space    =</span> <span class="st">"free"</span></span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">annotation =</span> annotation_labels)</span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>  theme_main</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-55-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting by species capped</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(capped_abundance_by_function) <span class="sc">+</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Geoms  ----------------</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>choanoflagellata_line <span class="sc">+</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> ncbi_name, <span class="at">y =</span> abundance, <span class="at">fill =</span> pathway_name, <span class="at">color =</span> <span class="fu">after_scale</span>(<span class="fu">darken</span>(fill, <span class="fl">0.1</span>)))</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">stat =</span> <span class="st">"identity"</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Labels  ---------------</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Espécies"</span>) <span class="sc">+</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Abundância média de proteínas em grupos ortólogos"</span>) <span class="sc">+</span></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylab("Average protein abundance in orthologous groups") +</span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Scales ----------------</span></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> tick_function, <span class="at">minor_breaks =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> annotation_colors <span class="sc">%&gt;%</span> <span class="fu">darken</span>(<span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Styling ---------------</span></span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a><span class="fu">facet_grid</span>(</span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>  pathway_name <span class="sc">~</span> clade_name</span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">scales   =</span> <span class="st">"free"</span></span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">space    =</span> <span class="st">"free"</span></span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">annotation =</span> annotation_labels)</span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>  theme_main</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-55-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="visualização-simplificada-abundância-por-clado" class="level4" data-number="1.17.2.3">
<h4 data-number="1.17.2.3" class="anchored" data-anchor-id="visualização-simplificada-abundância-por-clado"><span class="header-section-number">1.17.2.3</span> Visualização Simplificada: Abundância por Clado</h4>
<p>Para uma análise menos detalhada, é possível visualizar a abundância média de COGs apenas a nível de clado, ignorando as diferenças entre espécies individuais. Este gráfico fornece uma visão geral mais direta, ideal para identificar tendências globais.</p>
<ul>
<li>O eixo <strong>X</strong> representa os clados.</li>
<li>O eixo <strong>Y</strong> mostra a abundância média de proteínas em grupos ortólogos por clado.</li>
<li>As barras são agrupadas por clado e coloridas de acordo com as funções metabólicas (<em>pathway_name</em>).</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Ploting by clade</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(avg_abundance_by_function) <span class="sc">+</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> clade_name, <span class="at">y =</span> avg_abundance, <span class="at">fill =</span> pathway_name, <span class="at">color =</span> <span class="fu">after_scale</span>(<span class="fu">darken</span>(fill, <span class="fl">0.1</span>)))</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">stat =</span> <span class="st">"summary"</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">fun  =</span> <span class="st">"mean"</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> tick_function, <span class="at">minor_breaks =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> annotation_colors, <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>    pathway_name <span class="sc">~</span> .</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">scales   =</span> <span class="st">"free"</span></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">space    =</span> <span class="st">"free_y"</span></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">annotation =</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="st">""</span>, annotation_labels))</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Clados"</span>) <span class="sc">+</span></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Abundância média por clado"</span>) <span class="sc">+</span></span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a>  theme_average</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-56-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>